markhor_flippers/src/markhor_hw_interface_flippers.cpp

```diff
@@ -571,24 +580,21 @@ void MarkhorHWInterfaceFlippers::parseDrivePosition(std::string line)
 void MarkhorHWInterfaceFlippers::applyDrivePosition(std::unique_ptr<TalonSRX>& drive, float drive_position)
 {
   int error;
-  if (has_reset_event_occured_ == false)
```

Comment:
I understand that this was line was causing the estop bug, but do you know what it was meant to do? Maybe @lvanasse would be better suited to tell me.

Reply:
to be honest, I have no clue why it was there, prehaps a relic of an old architecture or something, makes no sense to me since applyDrivePosition can only be called from parseDrivePosition, which is called by readPositionFromFile which is called by loadDrivePosition, which is called by read() **only when a reset event occured**. The reset flag is reset by parseDrivePosition after it's first call, but it is called a total of 4 times per reset event so the first run (front left flipper) the flag is still true meaning it's position is not set and the 3 subsequent runs will work fine since the flag has been reset after the first.

Reply:
To be perfectly honest, it might be just an error on my part. Also, my memory is a bit fuzzy on that code base. But rereading the code, my first hypothesis would be that it is defensive programming. And make sure that we can apply the drive position only if there was a reset.